### ISSUE TITLE
Support for headers

### ISSUE THREAD

**Issue by kopcheski** [Amerousful/gatling-kafka#2]
That's something I could work around locally, but it would be nice to have it supported out of the box in the plugin.

In my scenario, I need to perform some "checks" on the ConsumerRecord headers. For example:

`.check(jsonPath("$.headers.header1").is("value1"))`

Of course, in order to be able to evaluate this json path, I had to extend the KafkaDsl, override the implicit function kafkaJsonPathCheckMaterializer and then manipulate the schema of the json to be evaluated. As you see above, the headers are in a attribute in the json root and the same goes for the original payload. Something like this:

`
{
 "payload" : { original_payload },
 "headers" : { "header1" : "value1", ... }
}
`

I don't think this would be an acceptable solution for the plugin itself, because whatever code already written using the jsonPath method will break, but I hope the example serves as food for thought.

How does that sound for you?

**Comment by Amerousful** [comment]
Unfortunately, it contradicts the main concept of checks in Gatling. All these checks, such as `jsonPath`, `xpath`, `substring`, and others, are designed for inspecting the response body. You can check it out in the [documentation.](https://gatling.io/docs/gatling/reference/current/core/check/#concepts)
As a particular case, header check has his own logic/methods for check: [documentation](https://gatling.io/docs/gatling/reference/current/http/check/#http-header) 

Another point to consider is that in Kafka, headers are represented as key-value pairs, and you cannot inspect them as json: [documentation](https://kafka.apache.org/20/javadoc/index.html?org/apache/kafka/connect/header/Header.html)

**Comment by kopcheski** [comment]
I see I haven't made my point clear, so let me try again. 😃 

I can't agree more on the reasons you exposed on why the approach I used it not ideal. I'm not suggesting to support headers as I have done either, as I said, that was just "food for thought".

Now, what I'm interested in discussing is whether you see value in supporting headers in the checks, as it is supported on plain Gatling when using http - as you pointed out - via `check(header().[is(), saveAs(), whatEverOtherCheckBuilder.Final()])`

So, does it sound like a valuable feature?

**Comment by Amerousful** [comment]
You really threw me off by emphasizing json path for headers so much, mentioning it repeatedly and even providing an example with `kafkaJsonPathCheckMaterializer` where you overrided the behavior/logic.  So, all the discussions were centered around that. 
I'm curious, could you please provide more details on how it actually worked when you check headers as json?

But overall, adding support for headers is a great idea.

**Comment by Amerousful** [comment]
More likely I'll added it in the same release with Java support

**Comment by kopcheski** [comment]
This is all that I've done:
ps: you'll quickly realize my limitations on Scala language.

1. Extending the plugin's behavior.
PredefExtended
```

object PredefExtended extends HeadersSupport {

}

/**
 * This class should be imported in favor of {@link io.github.amerousful.kafka.Predef}.
 */
class PredefExtended extends HeadersSupport {

}

trait HeadersSupport extends KafkaDsl {

  /**
   * Along with what is described in the {@link KafkaCheckMaterializerChild}, this is where the original code behavior
   * is changed. The original method's implementation is "overridden" by this implicit method, so the original schema
   * of the JSON object is changed. Formerly it would reflect the value of the {@link org.apache.kafka.clients.consumer.ConsumerRecord#value() V}.
   */
  override implicit def kafkaJsonPathCheckMaterializer
  (implicit jsonParsers: JsonParsers, configuration: GatlingConfiguration):
  CheckMaterializer[JsonPathCheckType, KafkaCheck, KafkaResponseMessage, JsonNode] =
    KafkaCheckMaterializerChild.payloadAndHeaderAsJson(jsonParsers)

}

```
KafkaCheckMaterializerForHeaders.scala
```
class KafkaCheckMaterializerChild[T, P] {

}

/**
 * It extends some functionalities of the original plugin's code by giving the ability to the jsonPath check to also
 * also inspect headers.
 */
object KafkaCheckMaterializerChild {

  /**
   * It creates a JSON object with two root level attributes; Content and Headers. The former contains the payload/message
   * of a ConsumerRecord whereas the latter contains a series a attribute/value pairs, one for each of the Headers.
   *
   * This is a breaking changing w.r.t the original plugin behavior, since it changes the schema of the json message
   * inspected via jsonCheck.
   */
  private def prepareHeadersAndPayload(jsonParsers: JsonParsers): Preparer[KafkaResponseMessage, JsonNode] =
    message => {
      val headers = prepareHeaders(message)
      val payloadAsJson = JsonMethods.parse(message.value()).asInstanceOf[JObject]
      val fullJson: JObject = ("Content", payloadAsJson) ~ ("Headers", headers)
      val plainJson = compact(JsonMethods.render(fullJson))
      jsonParsers.safeParse(plainJson)
    }

  /**
   * Fills in a map containing the headers of a ConsumerRecord.
   */
  private def prepareHeaders(message: KafkaResponseMessage): Map[String, String] =
  {
    val headerMap: Map[String, String] = message.headers().toArray.map {
      header => header.key() -> new String(header.value())
    }.toMap

    headerMap
  }

  def payloadAndHeaderAsJson(jsonParsers: JsonParsers): CheckMaterializer[JsonPathCheckType, KafkaCheck, KafkaResponseMessage, JsonNode] =
    new KafkaCheckMaterializer(prepareHeadersAndPayload(jsonParsers))

}
```

2. And now I make use of the "extended" behavior I added on my actual stress-test.
```
...
import io.github.amerousful.kafka.PredefExtended._
...
  def kafkaScenario(pluginName: String) = {
    kafka("Kafka: request with reply")
      .requestReply
      .topic("request-topic")
      .payload(myPayload())
      .replyTopic("reply-topic")
      .key("...")
      .header("header1", "value1")
      .check(jsonPath("$.Headers.header1").is("value1"))
```

**Comment by kopcheski** [comment]
Is this enough details to satisfy your curiosity?

**Comment by Amerousful** [comment]
@kopcheski Actually, I forgot that there's a way how to write check for headers in the current version. 
```scala
.check(simpleCheck {
      message => message.headers()....  % your condition %
    })
``` 

`simpleCheck` takes a function `KafkaResponseMessage => Boolean`, so you can operate with any record entity.

**Comment by Amerousful** [comment]
Headers check support was added in version 2.0
```scala
.check(
      header("headerKey1").is("headerValue1")
    )
```

**Comment by kopcheski** [comment]
That's exactly what I need.
